### PR TITLE
benchmark: add thanosbench tool

### PR DESCRIPTION
**Pull request by adamhosier**
Adds a temporary directory to hold the thanosbench tooling. This will be moved to its own repo in the future.
NOTE: the only difference between this and the standalone repo is the import paths for `thanosbench/pkg/tsdb`

### CODE REVIEW

**Comment by Bplotka** [comment]
Can you rebase?

**Comment by adamhosier** [comment]
@Bplotka yep, done

File: benchmark/Gopkg.toml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+[[constraint]]
+  name = "github.com/pkg/errors"
+  version = "0.8.0"
+
+[[constraint]]
+  name = "k8s.io/client-go"
```
Comment (Bplotka) [inline benchmark/Gopkg.toml]: seems like you cannot resolve this version with Prometheus. Can you ensure you can run `make deps` first?

File: Gopkg.toml [hunk 2]
```diff
@@ -1,3 +1,5 @@
+ignored = ["github.com/improbable-eng/thanos/benchmark/*"]
```
Comment (Bplotka) [inline Gopkg.toml]: Adam you are true magician, nice finding.

File: benchmark/README.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
+# Thanosbench
+A series of benchmarking/stress test tools for [thanos](https://github.com/improbable-eng/thanos).
```
Comment (Bplotka) [inline benchmark/README.md]: Let's put some link in Thanos main README as well (: What do you think?

**Comment by Bplotka** [comment]
+ rebase

**Comment by Bplotka** [comment]
Ignore CI, fixing now

**Pull request merged**